datasquare_test.go

```diff
@@ -471,6 +471,57 @@ func BenchmarkEDSRootsWithErasuredNMT(b *testing.B) {
 	}
 }
 
+func BenchmarkEDSRootsWithBufferedErasuredNMT(b *testing.B) {
+	const mebibyte = 1024 * 1024            // bytes
+	ODSSizeByteUpperBound := 512 * mebibyte // converting 512 MiB to bytes
+	totalNumberOfShares := float64(ODSSizeByteUpperBound) / shareSize
+	// the closest power of 2 of the square root of
+	// the total number of shares
+	nearestPowerOf2ODSSize := math.Pow(2, math.Ceil(math.Log2(math.Sqrt(
+		totalNumberOfShares))))
+	namespaceIDSize := 8
+
+	for squareSize := 32; squareSize <= int(nearestPowerOf2ODSSize); squareSize *= 2 {
+		// number of shares in the original data square's row/column
+		odsSize := squareSize
+		// number of shares in the extended data square's row/column
+		edsSize := 2 * odsSize
+		// generate an EDS with edsSize X edsSize dimensions in terms of shares.
+		// the generated EDS does not conform to celestia-app specs in terms
+		// of namespace version, also no erasure encoding takes place
+		// yet none of these should impact the benchmarking
+		ds := genRandSortedDS(edsSize, shareSize, namespaceIDSize)
+
+		// a tree constructor for erasured nmt
+		parallelOps := 40
+		treeConstructor := newBufferedErasuredNamespacedMerkleTreeConstructor(uint64(edsSize), parallelOps,
+			nmt.NamespaceIDSize(namespaceIDSize), nmt.IgnoreMaxNamespace(true),
+			nmt.InitialCapacity(odsSize*2))
+
+		square, err := newDataSquare(ds, treeConstructor, shareSize)
+		if err != nil {
+			b.Errorf("Failure to create square of size %d: %s", odsSize, err)
+		}
+		square.setParallelOps(parallelOps)
+		// the total size of the ODS in MiB
+		odsSizeMiBytes := odsSize * odsSize * shareSize / mebibyte
+		// the total size of the EDS in MiB
+		edsSizeMiBytes := 4 * odsSizeMiBytes
+		b.Run(
+			fmt.Sprintf("%dx%dx%d ODS=%dMB, EDS=%dMB", odsSize, odsSize,
+				int(square.shareSize),
+				odsSizeMiBytes, edsSizeMiBytes),
+			func(b *testing.B) {
+				for n := 0; n < b.N; n++ {
+					square.resetRoots()
+					err := square.computeRoots()
```

Comment:
can we benchmark the times to create the tree and also the time to compute the root separately? it would be good to know how much each take

Reply:
only what you listed, but just to make sure that's really the case. It's not blocking tho